x-pack/plugins/observability_solution/observability_ai_assistant/server/service/index.ts

```diff
@@ -43,65 +42,56 @@ export const resourceNames = {
     conversations: getResourceName('index-template-conversations'),
     kb: getResourceName('index-template-kb'),
   },
-  pipelines: {
-    kb: getResourceName('kb-ingest-pipeline'),
-  },
 };
 
 export class ObservabilityAIAssistantService {
   private readonly core: CoreSetup<ObservabilityAIAssistantPluginStartDependencies>;
   private readonly logger: Logger;
-  private readonly getModelId: () => Promise<string>;
-  public kbService?: KnowledgeBaseService;
+  private readonly getSearchConnectorModelId: () => Promise<string>;
+  private kbService?: KnowledgeBaseService;
   private enableKnowledgeBase: boolean;
 
   private readonly registrations: RegistrationCallback[] = [];
 
   constructor({
     logger,
     core,
-    taskManager,
-    getModelId,
+    getSearchConnectorModelId,
     enableKnowledgeBase,
   }: {
     logger: Logger;
     core: CoreSetup<ObservabilityAIAssistantPluginStartDependencies>;
-    taskManager: TaskManagerSetupContract;
-    getModelId: () => Promise<string>;
+    getSearchConnectorModelId: () => Promise<string>;
     enableKnowledgeBase: boolean;
   }) {
     this.core = core;
     this.logger = logger;
-    this.getModelId = getModelId;
+    this.getSearchConnectorModelId = getSearchConnectorModelId;
     this.enableKnowledgeBase = enableKnowledgeBase;
 
-    this.allowInit();
-  }
-
-  getKnowledgeBaseStatus() {
-    return this.init().then(() => {
-      return this.kbService!.status();
-    });
+    this.registerInit();
   }
 
   init = async () => {};
 
-  private allowInit = () => {
+  private registerInit = () => {
     this.init = once(async () => {
       return this.doInit().catch((error) => {
-        this.allowInit();
+        this.registerInit(); // reset the once flag if an error occurs
         throw error;
       });
     });
   };
 
   private doInit = async () => {
     try {
-      const [coreStart, pluginsStart] = await this.core.getStartServices();
+      this.logger.info('Setting up index assets');
+      const [coreStart] = await this.core.getStartServices();
 
-      const elserModelId = await this.getModelId();
+      const esClient = {
+        asInternalUser: coreStart.elasticsearch.client.asInternalUser,
+      };
```

Comment:
It's because we are passing `esClient` to `KnowledgeBaseService` ([see](https://github.com/sorenlouv/kibana/blob/657c82a4901d714257561e2c33b37f884a4ad39f/x-pack/plugins/observability_solution/observability_ai_assistant/server/service/index.ts#L172)) but I agree it looks weird and will change